### DISCUSSION TITLE
Where clause on one field but sort on another.

### DISCUSSION THREAD

**Discussion by aidevani** [isar/isar#1502] [Help] [answered]
I am trying to get the **last item** of a collection where a certain field is false. My Isar collection is of Message objects with fields like messageContent and isSentByMe (bool) and the Isar Id which is auto-increment. I have an index on isSentByMe. I am trying to get the last message that was not sent by me. Since I have an index on isSentByMe, I tried  dbref.messages.**where(sort: Sort.desc).isSentByMeEqualTo(false).limit(1).findFirst()**. 

But its giving me the first message even though sort in where clause is set to Sort.desc. Any help on how I can get the **last** message that was not sent by me?

**Comment by Jtplouffe** [comment, marked as answer]
In your query, the `where(sort: Sort.desc)` will sort by the index that was used, in that case, `isSentByMe`.
You will need to do an explicit sort on the timestamp at which the record was inserted.
